### ISSUE TITLE
[INFRA] add a GH workflow to regularly update pre-commit hooks

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#3490]
Done.

Once this is merged I will open a separate PR with a workflow to check and update the hooks version weekly so we don't have to remember to regularly update this.

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/3484#discussion_r1098683579_
